pkg/controllers/cassandra/nodepool/resource.go

```diff
@@ -35,6 +39,17 @@ func StatefulSetForCluster(
 	statefulSetName := util.NodePoolResourceName(cluster, np)
 	seedProviderServiceName := util.SeedProviderServiceName(cluster)
 	nodePoolLabels := util.NodePoolLabels(cluster, np.Name)
+
+	datacenter := cassDefaultDatacenter
+	if np.Datacenter != "" {
+		datacenter = np.Datacenter
+	}
+
+	rack := np.Name
+	if np.Rack != "" {
+		rack = np.Rack
+	}
+
```

Comment:
:grey_question: Does Cassandra place any limit on the length of or characters in  this field? `np.Name` can be any DNS label (I think).

Reply:
From what I can tell from cassandra's source, the maximum is the maximum size of a java string (big)